### ISSUE TITLE
Flag potential for optimized `rep()` calls

### ISSUE THREAD

**Issue by Bisaloo** [r-lib/lintr#2898]
It's quite common to see concatenated `rep()` calls, when it could be turn into a single one:

``` r
microbenchmark::microbenchmark(
  c(rep("a", 3000), rep("A", 5000)),
  rep(c("a", "A"), c(3000, 5000)),
  check = "identical",
  times = 1e4
)
#> Unit: microseconds
#>                               expr    min      lq     mean  median     uq
#>  c(rep("a", 3000), rep("A", 5000)) 53.935 61.3530 83.01887 66.7485 90.372
#>    rep(c("a", "A"), c(3000, 5000)) 28.601 32.4185 43.44523 35.2165 47.762
#>       max neval
#>  3012.145 10000
#>  3303.804 10000
```

<sup>Created on 2025-07-26 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

Is this a case of `inner_combine_linter()` or should it be merged with `rep_len_linter()` into a broader `rep_linter()`?

**Comment by MichaelChirico** [comment]
> Is this a case of `inner_combine_linter()` or should it be merged with `rep_len_linter()` into a broader `rep_linter()`?

I share the exact question!

I also see code like this a lot, but often skip "correcting" it -- it's usually in tests where performance is not ~all that important, and it's not always a clear readability improvement.

I would suggest testing a proposal here out on some code bases to get to a happy medium where this is often providing strong signal for code that can be improved vs. potentially "annoying" large # of lints for test code that already "just works".

**Comment by AshesITR** [comment]
If we do this and statically detect the same expr for times, we can recommend `rep(c("a", "A"), each = x)`.

Regarding scope, this feels more tied to `rep()` than to `c()` so I'm leaning towards `rep_linter()`.
